capa/capabilities/dynamic.py

```diff
@@ -51,11 +55,11 @@ def find_call_capabilities(
 
 def find_thread_capabilities(
     ruleset: RuleSet, extractor: DynamicFeatureExtractor, ph: ProcessHandle, th: ThreadHandle
-) -> tuple[FeatureSet, MatchResults, MatchResults]:
+) -> tuple[FeatureSet, MatchResults, MatchResults, MatchResults]:
```

Comment:
maybe we should introduce a dataclass/namedtuple for these return values. they're hard to keep straight.

Reply:
this PR already has breaking changes, so this might be a good place to add them.

Reply:
yeah, good idea, but not a blocker here